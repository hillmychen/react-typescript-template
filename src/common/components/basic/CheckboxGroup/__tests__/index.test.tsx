import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { CheckboxGroup } from '../index'

test('CheckboxGroup 选中：男', () => {
  const { asFragment, getByText } = render(
    <CheckboxGroup
      value={[1]}
      dataSource={[
        { label: '男', value: 1 },
        { label: '女', value: 2 },
        { label: '未知', value: 3, disabled: true },
      ]}
    />
  )

  expect(getByText('男').closest('label')).toHaveClass('ant-checkbox-wrapper-checked')
  expect(getByText('女').closest('label')).not.toHaveClass('ant-checkbox-wrapper-checked')
  expect(getByText('未知').closest('label')).toHaveClass('ant-checkbox-wrapper-disabled')

  expect(asFragment()).toMatchSnapshot()
})
