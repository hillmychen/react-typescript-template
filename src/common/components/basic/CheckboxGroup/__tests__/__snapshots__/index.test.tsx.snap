// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup 选中：男 1`] = `
<DocumentFragment>
  <div
    class="ant-checkbox-group checkbox"
  >
    <label
      class="ant-checkbox-wrapper ant-checkbox-wrapper-checked ant-checkbox-group-item"
    >
      <span
        class="ant-checkbox ant-checkbox-checked"
      >
        <input
          checked=""
          class="ant-checkbox-input"
          type="checkbox"
          value="1"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        男
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item"
    >
      <span
        class="ant-checkbox"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="2"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        女
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled ant-checkbox-group-item"
    >
      <span
        class="ant-checkbox ant-checkbox-disabled"
      >
        <input
          class="ant-checkbox-input"
          disabled=""
          type="checkbox"
          value="3"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        未知
      </span>
    </label>
  </div>
</DocumentFragment>
`;
